backend/lucene/src/main/java/org/hibernate/search/backend/lucene/analysis/LuceneAnalyzerDescriptor.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.analysis;
+
+import org.hibernate.search.engine.backend.analysis.AnalyzerDescriptor;
+import org.hibernate.search.util.common.annotation.Incubating;
+
+import org.apache.lucene.analysis.Analyzer;
+
+/**
+ * The Lucene backend-specific extension of the analyzer descriptor.
+ */
+@Incubating
+public interface LuceneAnalyzerDescriptor extends AnalyzerDescriptor {
```

Comment:
I couldn't find integration tests for this... How do you intend to expose it to users? Ask them to cast? :/

Unless you intend to provide some `.extension` method on the index descriptor, I'd say those interfaces should remain implementation for now (or disappear if they are not useful).